index.bs

```diff
@@ -795,6 +795,128 @@ the following list:
+  undefined close(optional WebSocketCloseInfo closeInfo);
+};
+</xmp>
+
+A {{WebSocketStream}} object has an associated <dfn>url</dfn> (a [=URL record=]).
+
+A {{WebSocketStream}} object has an associated <dfn>opened promise</dfn>.
+
+A {{WebSocketStream}} object has an associated <dfn>closed promise</dfn>.
+
+A {{WebSocketStream}} object has an associated <dfn>was ever connected flag</dfn>, which is
+initially unset.
+
+<dl class="domintro non-normative">
+ : <code>|socket| = new {{WebSocketStream/constructor(url, options)|WebSocketStream}}(|url| [, |options| ]</code>
+ :: Creates a new {{WebSocketStream}} object, immediately establishing the associating WebSocket
+  connection.
+
+  |url| is a string giving the <a dfn spec=url>URL</a> over which the connection is established.
+  Only "`ws`" or "`wss`" schemes are allowed; others will cause a "{{SyntaxError}}"
+  {{DOMException}}. URLs with [=fragments=] will also cause such an exception.
+
+  The |options| argument is an object whose properties can be set as follows:
+
+  : {{WebSocketStreamOptions/protocols}}
+  :: An array of strings. If it is omitted, it is equivalent to an empty array. Each string in the
+   array is a subprotocol name. The connection will only be established if the server reports that
+   it has selected one of these subprotocols. The subprotocol names have to match the requirements
+   for elements that comprise the value of \`<a http-header>`Sec-WebSocket-Protocol`</a>\` fields as
+   defined by The WebSocket protocol.  [[!WSP]]
+
+  : {{WebSocketStreamOptions/signal}}
+  :: An {{AbortSignal}} that can be used to abort the handshake. After the handshake is complete the
+   signal does nothing.
+
+ : <code>|socket| . {{WebSocketStream/url}}</code>
+ :: Returns the <a lt="url">URL that was used</a> to establish the WebSocket connection.
+
+ : <code>|socket| . {{WebSocketStream/opened}}</code>
+ :: Returns a {{promise}} which resolves when the handshake successfully completes, or rejects if
+  the handshake fails. On success, it resolves to an object with the following properties:
+
+  : {{WebSocketOpenInfo/readable}}
+  :: A {{ReadableStream}} that can be used to read messages from the server. Each chunk read
+   corresponds to one message. Text messages will be read as strings; binary messages will be read
+   as ArrayBuffer objects.
+
+   The stream can be closed by calling {{ReadableStream/cancel()}} on
+   {{WebSocketOpenInfo/readable}}. If the argument passed to {{ReadableStream/cancel()}} is an
+   object with a {{WebSocketCloseInfo/code}} property and an <span class=allow-2119>optional</span> {{WebSocketCloseInfo/reason}}
+   property then {{WebSocketCloseInfo/code}} will be used as [=the WebSocket connection close
+   code=] and {{WebSocketCloseInfo/reason}} or the empty string will be used as [=the WebSocket
+   connection close reason=].
+
+   If no messages are read, or if messages are read slower than they are sent, then backpressure
```

Comment:
Is there a backpressure support for writable stream?

from this explainer looks like there should be a back pressure support from writable stream. Do you think we should explicitly mention it?

https://github.com/ricea/websocketstream-explainer/blob/master/README.md

Reply:
I added a note that backpressure applies. PTAL.